src/fides/api/schemas/connection_configuration/connection_secrets_s3.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List, Optional
+
+from pydantic import Field, root_validator
+
+from fides.api.schemas.base_class import NoValidationSchema
+from fides.api.schemas.connection_configuration.connection_secrets import (
+    ConnectionConfigSecretsSchema,
+)
+from fides.api.schemas.storage.storage import AWSAuthMethod
+
+
+class S3Schema(ConnectionConfigSecretsSchema):
+    """Schema to validate the secrets needed to connect to Amazon S3"""
+
+    auth_method: AWSAuthMethod = Field(
+        title="Authentication Method",
+        description="Determines which type of authentication method to use for connecting to Amazon S3",
+    )
+
+    aws_access_key_id: Optional[str] = Field(
+        title="Access Key ID",
+        description="Part of the credentials that provide access to your AWS account. This is required if using secret key authentication.",
+    )
+    aws_secret_access_key: Optional[str] = Field(
+        title="Secret Access Key",
+        description="Part of the credentials that provide access to your AWS account. This is required if using secret key authentication.",
+        sensitive=True,
+    )
+
+    aws_assume_role_arn: Optional[str] = Field(
```

Comment:
hm I never did get this to work in testing, just using the user urn. in the sandbox creds

Reply:
I don't know much about ARN's, reading here: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html, can you explain/point to good docs about what this particular param is for?

Reply:
err yeah so the use case here is to use a _different_ ARN whose role you want to assume, using the provided credentials as a "starting" user. 

honestly i'm not sure the best reading for this - it's a concept i learned about a little while ago, and @daveqnet refreshed my memory nicely [here](https://ethyca.slack.com/archives/C01TUEAGCR1/p1718027044938379?thread_ts=1718025829.889269&cid=C01TUEAGCR1), which i think helps explain the use case? 

i can pass along in slack the role ARN that i'm assuming, as a test! it's basically a standard "s3 full access" role.

Reply:
thanks for the info 👍